--- a/net/minecraft/world/entity/projectile/ProjectileUtil.java
+++ b/net/minecraft/world/entity/projectile/ProjectileUtil.java
@@ -203,10 +_,12 @@
                     if (!optional1.isEmpty()) {
                         Vec3 vec3 = optional1.get();
                         Vec3 center = boundingBox1.getCenter();
+                        if (!level.paperConfig().sugarcane.projectiles.allowSurfaceHitsThroughBlocks) { // Sugarcane - restore pre-1.21.11 margin hits through blocks
                         BlockHitResult blockHitResult = level.clipIncludingBorder(new ClipContext(vec3, center, clipType, ClipContext.Fluid.NONE, source));
                         if (blockHitResult.getType() != HitResult.Type.MISS) {
                             center = blockHitResult.getLocation();
                         }
+                        } // Sugarcane - restore pre-1.21.11 margin hits through blocks
 
                         Optional<Vec3> optional2 = entity.getBoundingBox().clip(vec3, center);
                         if (optional2.isPresent()) {
